stk/utilities/utilities.py

```diff
@@ -1143,10 +1146,371 @@ def vector_theta(vector1, vector2):
     """
 
     numerator = np.dot(vector1, vector2)
-    denominator = (np.linalg.norm(vector1) *
-                   np.linalg.norm(vector2))
+    denominator = (np.linalg.norm(vector1) * np.linalg.norm(vector2))
     # This if statement prevents returns of NaN due to floating point
     # incurracy.
     if np.isclose(numerator, denominator, atol=1e-8):
         return 0.0
     return np.arccos(numerator/denominator)
+
+
+class XTBInvalidSolventError(Exception):
+    ...
+
+
+def is_valid_xtb_solvent(gfn_version, solvent):
+    """
+    Check if solvent is valid for the given GFN version.
+
+    Parameters
+    ----------
+    gfn_version : :class:`int`
+        GFN parameterization version. Can be: ``0``, ``1`` or ``2``.
+
+    solvent : :class:`str`
+        Solvent being tested.
+
+    Returns
+    -------
+    :class:`bool`
+        ``True`` if solvent is valid.
+
+    See Also
+    --------
+    # https://xtb-docs.readthedocs.io/en/latest/gbsa.html
+
+    """
+    if gfn_version == 0:
+        return False
+    elif gfn_version == 1:
+        valid_solvents = set([
+            'acetone', 'acetonitrile', 'benzene',
+            'CH2Cl2'.lower(), 'CHCl3'.lower(), 'CS2'.lower(),
+            'DMSO'.lower(), 'ether', 'H2O'.lower(),
+            'methanol', 'THF'.lower(), 'toluene'
+        ])
```

Comment:
Dont make a set from a list, because you are still making the list first, which comes at both a computational and memory cost, You can create a set directly with
```python
        valid_solvents = {
            'acetone', 'acetonitrile', 'benzene',
            'CH2Cl2'.lower(), 'CHCl3'.lower(), 'CS2'.lower(),
            'DMSO'.lower(), 'ether', 'H2O'.lower(),
            'methanol', 'THF'.lower(), 'toluene'
        }

```

Reply:
Those are curly brackets (its not obvious on my github). bascially curly brackets where values are separated by commas and ":" is not present means set. Ofc if you add ":" you make a dict. {} defaults to empty dict, so if you want to make an empty set use "set()"